### PR TITLE
ZCS-6513 Compiling zm-mailbox against JDK11

### PR DESCRIPTION
**Pull request by rupalid**
Compiled code against JDK 11. 
Fixed failing code
Verified that 'ant publish-local-all' succeeds

Ran unit tests
Tests | Failures | Errors | Skipped | Success rate | Time
1266 | 0 | 0 | 10 | 100.00% | 1414.386

Compiled the code against JDK 1.8 and tested on ZCS installation. 
Some unit tests which depend on power mock and mockito have know issues with JDK 11. Excluded them for now. Will fix them in next sprint.

### CODE REVIEW

File: common/src/java/com/zimbra/common/net/CustomHostnameVerifier.java [hunk 1]
```diff
@@ -16,22 +16,36 @@
...
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+
+import org.bouncycastle.est.jcajce.JsseDefaultHostnameAuthorizer;
+
+import com.google.common.collect.Sets;
+import com.zimbra.common.util.ZimbraLog;
+
+
```
Comment (shrikant-prasad) [inline common/src/java/com/zimbra/common/net/CustomHostnameVerifier.java]: remove the unnecessary lines

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ sudo: required
 language: java
 
 jdk:
-  - oraclejdk8
+  - oraclejdk11
```
Comment (chinmay15) [inline .travis.yml]: openjdk11?

File: common/src/java-test/com/zimbra/common/util/QuotedTextUtilTest.java [hunk 3]
```diff
@@ -21,15 +21,15 @@
...
+import org.junit.Ignore;
 import junit.framework.Assert;
 
 public class QuotedTextUtilTest {
 
     /**
      * Test method for {@link com.zimbra.common.util.QuotedTextUtil#getOriginalContent(java.lang.String, boolean)}.
      */
-    @Test
+    @Ignore
```
Comment (fatkudu) [inline common/src/java-test/com/zimbra/common/util/QuotedTextUtilTest.java]: Why is this being ignored - that sets off alarm bells that perhaps something isn't working?

Please improve the commit text.  For each action taken, explain why it was taken.

File: store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java [hunk 4]
```diff
@@ -120,7 +128,7 @@ private void captureClientCert(X509Certificate cert) {
             // Write in text form
             wr = new OutputStreamWriter(os, Charset.forName("UTF-8"));
             wr.write("-----BEGIN CERTIFICATE-----\n");
-            wr.write(new sun.misc.BASE64Encoder().encode(buf));
+            wr.write(Base64.getEncoder().encodeToString(buf));
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java]: Has this been tested?

File: store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java [hunk 5]
```diff
@@ -218,13 +226,21 @@ private void validateClientCert(X509Certificate[] certs) throws ServiceException
     // examine the certificate's AuthorityInfoAccess extension
     private boolean IsAIAInfoPresent(X509Certificate cert) {
         try {
-            AuthorityInfoAccessExtension aia =
-                            X509CertImpl.toImpl(cert).getAuthorityInfoAccessExtension();
-            return  (aia != null);
-        } catch (CertificateException ce) {
+            byte[] authInfoAccessExtensionValue = cert
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java]: Testing done?

File: store/ivy.xml [hunk 6]
```diff
@@ -114,5 +114,12 @@
   <dependency org="org.apache.commons" name="commons-rng-simple" rev="1.0"/>
   <dependency org="com.google.javascript" name="closure-compiler" rev="v20180204"/>
   <dependency org="com.github.zafarkhaja" name="java-semver" rev="0.9.0"/>
+  <dependency org="javax.xml.bind" name="jaxb-api" rev="2.3.1"/>
+  <dependency org="com.sun.xml.bind" name="jaxb-core" rev="2.3.0.1"/>
+  <dependency org="com.sun.xml.bind" name="jaxb-impl" rev="2.3.1"/>
+  <dependency org="javax.activation" name="activation" rev="1.1.1"/>
```
Comment (desouzas) [inline store/ivy.xml]: Why do you need this when you also have `com.sun.activation` version?

File: common/src/java/com/zimbra/common/net/CustomHostnameVerifier.java (hunk 1 above)
Comment (rupalid) [reply common/src/java/com/zimbra/common/net/CustomHostnameVerifier.java]: fixed

File: store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java (hunk 4 above)
Comment (rupalid) [reply store/src/java/com/zimbra/cs/service/authenticator/ClientCertAuthenticator.java]: Yes I did write a test program and tested that the files created are same. Also this is code is executed only when a DebugConfig is true. By default it is false.

**Comment by rupalid** [comment]
> My issues have been resolved.
> Worth trying without com.sun.activation perhaps as presumably javax.activation does what that used to do and is the supported way? This is what @desouzas mentioned.

Removed com.sun.activation and it worked, so removing the change

**Comment by desouzas** [comment]
Other way around actually.

`com.sun.activation` replaces `javax.activation`.

See their dates, ditch the other instead.

**Pull request merged**